### DISCUSSION TITLE
Overpayments processing

### DISCUSSION THREAD

**Discussion by divinationstar** [btcpayserver/btcpayserver#2459] [Ideas / Feature Requests]
There should be a setting in "server settings", "general settings", "payment" section that says "Automatically credit overpayment amount". yes/no. If set to yes, then amount paid equals (invoice amount + overpayment amount). If no, then no change at all. 

If the user wants a refund then overpayment amount can be in "issue refund" process by default or a choice of what to refund (total amount or overpayment amount) can be selected instead.

**Comment by Kukks** [comment]
Please describe this more clearly with examples. I'm not fully sure what this feature would be

**Reply by divinationstar** [reply]
The problem is when someone makes an overpayment to an invoice. Currently, you can either ignore it or make a refund. Neither of these is desirable. What I have to do when a user makes an overpayment is manually process the overpayment. 

First I need to manually calculate the overpayment which is BTC price X BTC overpayment amount. Then I need to go into my other system and manually credit the user account for the total overpayment amount. This manual work can be streamlined with an addition to BTCpayserver system. 

The solution is as described in my initial message.
